inference-engine/src/transformations/src/transformations/convert_opset1_to_legacy/convert_proposal_to_proposal_ie.cpp

```diff
@@ -8,58 +8,78 @@
 #include <vector>
 
 #include <ngraph/opsets/opset1.hpp>
-
+#include <ngraph/opsets/opset4.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
 #include <ngraph_ops/proposal_ie.hpp>
 #include <ngraph/rt_info.hpp>
 
-ngraph::pass::ConvertProposalToLegacyMatcher::ConvertProposalToLegacyMatcher() {
-    auto input_0 = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1, 1, 1});
-    auto input_1 = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1, 1, 1});
-    auto input_2 = std::make_shared<pattern::op::Label>(element::f32, Shape{3});
+bool convert_to_proposal_ie(std::shared_ptr<ngraph::op::v0::Proposal> proposal, bool add_second_out = false) {
+    ngraph::Output<ngraph::Node> last;
 
-    ngraph::op::ProposalAttrs attr = {};
+    ngraph::NodeVector ops_to_replace, new_ops;
+    ops_to_replace.push_back(proposal);
 
-    auto proposal = std::make_shared<ngraph::opset1::Proposal>(input_0, input_1, input_2, attr);
+    if (auto reshape = std::dynamic_pointer_cast<ngraph::opset1::Reshape>(
+            proposal->input_value(2).get_node_shared_ptr())) {
+        auto input_shape = reshape->get_input_shape(0);
+        if (input_shape.size() == 2) {
+            last = reshape->input_value(0);
+            ops_to_replace.push_back(reshape);
+        }
+    }
+
+    if (!last.get_node_shared_ptr()) {
+        std::vector<int64_t> dims{1, -1};
+        auto const_shape = std::make_shared<ngraph::opset1::Constant>(ngraph::element::i64, ngraph::Shape{2}, dims);
+        last = std::make_shared<ngraph::opset1::Reshape>(proposal->input_value(2), const_shape, true);
+        new_ops.push_back(last.get_node_shared_ptr());
+    }
+
+    auto proposal_ie = std::make_shared<ngraph::op::ProposalIE>(proposal->input_value(0),
+                                                                proposal->input_value(1),
+                                                                last,
+                                                                proposal->get_attrs());
+    if (add_second_out) {
+        proposal_ie->set_output_type(1, proposal->get_input_element_type(0), ngraph::Shape {1});  // validate will set true shape
```

Comment:
@GlebKazantaev , do we really need this line and the next one? Can we achieve the same result somehow differently?

Reply:
I did this in order to reduce code dublication. If we want to set shape right here we need to check again if it's dynamic or static but this was already done in `validate_and_infer_types` of `Proposal` and `proposal_ie`. But let't wait what @GlebKazantaev think about this.

Reply:
I suggested to parameterize ProposallE and in this function you will pass this argument (for example `add_second_output` or smth more meaningful) to ProposalIE ctor

Reply:
Done.